### PR TITLE
Handle bilinear caching in Pyresample

### PR DESCRIPTION
**Pull request by pnuu**
This is the other half of moving bilinear resampler caching to Pyresample, the other bit is [here](https://github.com/pytroll/pyresample/pull/303). The PR in Pyresample needs to be merged and released, then retrigger the tests here and make a new Satpy release ASAP to have a matching features available.

 - [x] Tests added 
 - [x] Passes ``flake8 satpy``

### CODE REVIEW

**Review by mraspaud — APPROVED** [review]
LGTM

**Comment by mraspaud** [comment]
This needs a new pyresample release, right?

File: satpy/resample.py [hunk 1]
```diff
@@ -145,10 +145,12 @@
...
 try:
     from pyresample.resampler import BaseResampler as PRBaseResampler
-    from pyresample.gradient import GradientSearchResampler
 except ImportError:
-    warnings.warn('Gradient search resampler not available, upgrade Pyresample.')
     PRBaseResampler = None
+try:
+    from pyresample.gradient import GradientSearchResampler
+except ImportError:
+    warnings.warn('Gradient search resampler not available. Maybe missing `shapely`?.')
```
Comment (djhoese) [inline satpy/resample.py:153]: Is it possible to do this check inside some other function or to only show it when a user requests gradient search? It gets really annoying when most people probably aren't going to use the gradient search (for now at least).

Comment (pnuu) [reply satpy/resample.py]: Moved the warning to a place where the user tries to use it.

**Review by djhoese — APPROVED** [review]
Thanks!

**Comment by pnuu** [comment]
I added a quick'n'dirty fix to `yaml_reader` tests that failed due to https://github.com/pytroll/pyresample/pull/290 making `area_extent` immutable.

**Review by sfinkens — APPROVED** [review]
Nice job!

**Pull request merged**